### PR TITLE
feat: DataVpc construct

### PR DESCRIPTION
**Pull request by vgkowski**
**Issue #, if available:**

## Description of changes:

Add a new construct for creating a standard VPC for data platforms including VPC flow logs in CLoudWatch log group and S3 VPC endpoint

**Checklist**

 

* [x] Update tests
* [ ] Update docs
* [x] PR title follows [conventional commit semantics](https://www.conventionalcommits.org/en/v1.0.0/) (`fix: `, `feat: `, `docs: `, ...)

## Breaking change checklist

**RFC issue #**:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by vgkowski** [comment]
I have ran the e2e test that failed because of exceeded quota and it passed

File: framework/src/utils/lib/data-vpc-props.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  /**
+   * The KMS key for the VPC flow log group
+   * @default - A new KMS key is created
+   */
+  readonly flowLogKey?: IKey;
+  /**
+   * The IAM role for the VPC flow log
+   * @default - A new IAM role is created
+   */
+  readonly flowLogRole?: IRole;
```
Comment (lmouhib) [inline framework/src/utils/lib/data-vpc-props.ts:27]: can you please update to specify which principal should be supplied to the role for it to be assumed correctly by AWS?

https://docs.aws.amazon.com/vpc/latest/userguide/flow-logs-cwl.html#flow-logs-iam-role

File: framework/src/utils/lib/data-vpc-props.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+export interface DataVpcProps {
+  /**
+   * The CIDR to use to create the subnets in the VPC.
+   */
+  readonly vpcCidr: string;
+  /**
+   * The KMS key for the VPC flow log group
+   * @default - A new KMS key is created
+   */
```
Comment (lmouhib) [inline framework/src/utils/lib/data-vpc-props.ts:21]: Can you add a note to ask the user to add a resource policy to the key to allow its use by the flow log?

https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/encrypt-log-data-kms.html

File: framework/src/utils/lib/data-vpc-props.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  readonly flowLogRole?: IRole;
+  /**
+   * The retention period to apply to VPC Flow Logs
+   * @default - One week retention
+   */
+  readonly flowLogRetention?: RetentionDays;
+  /**
+   * The policy to apply when the bucket is removed from this stack.
+   * @default - RETAIN (The bucket will be orphaned).
```
Comment (lmouhib) [inline framework/src/utils/lib/data-vpc-props.ts:35]: remove this or change it to log "(The bucket will be orphaned)"

File: framework/src/utils/lib/data-vpc.ts [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+  /**
+   * The IAM role used to publish VPC Flow Logs
+   */
+  public readonly flowLogRole: IRole;
+  /**
+   * The CloudWatch Log Group created for the VPC flow logs
+   */
+  public readonly flowLogGroup: ILogGroup;
+  /**
+   * The S3 VPC endpoint
```
Comment (lmouhib) [inline framework/src/utils/lib/data-vpc.ts:33]: VPC endpoint gateway

**Review by lmouhib — CHANGES_REQUESTED** [review]
few minor comments

**Review by lmouhib — APPROVED** [review]
LGTM

**Pull request merged**